.github/workflows/tier_one_coverage.yml

```diff
@@ -0,0 +1,77 @@
+name: TierOneCoverage
+
+on: workflow_dispatch
+
+jobs:
+  check-coverage:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Setup Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.9'
+
+      - name: Make Temp Dir for Artifacts
+        run: |
+          pwd
+          mkdir coverageDir 
+          touch coverageDir/coverage.csv
+
+      - name: Run Coverage on Tier One Plugins
+        env:
+          TIER_ONE_PLUGINS_ICON: math active_directory_ldap advanced_regex microsoft_atp_safe_links base64 string ssh basename subnet microsoft_teams csv python_3_script type_converter rapid7_insight_agent datetime rapid7_insightidr whois dig rapid7_insightvm rapid7_intsights extractit rapid7_vulndb rapid7_insightappsec hashit rest rapid7_insights html servicenow jira sleep
+
+        run: |
+          set +e
+          cd plugins
+          for plugin in $TIER_ONE_PLUGINS_ICON
+          do
+            cd $plugin
+            echo $plugin
+            workingDir=$(pwd)
+            
+            # Creating venv so we can install required libraries and later delete them to avoid conflicts
```

Comment:
we experimented with this approach but found that there were too many clashes with dependancies unfortunately.
After discussing with @dlaverty-r7 we decided that this was the optimal way forward. We agreed that this script does not need to be super efficient as it will only be run once a month.